docs/guide/concept-autoloading.md

```diff
@@ -2,93 +2,75 @@ Class Autoloading
-either `@foo` or `@foo/bar` must be a [root alias](concept-aliases.md#defining-aliases).
-
-When using the [Basic Project Template](start-installation.md), you may put your classes under the top-level
-namespace `app` so that they can be autoloaded by Yii without the need of defining a new alias. This is because
-`@app` is a [predefined alias](concept-aliases.md#predefined-aliases), and a class name like `app\components\MyClass`
-can be resolved into the class file `AppBasePath/components/MyClass.php`, according to the algorithm just described.
-
-In the [Advanced Project Template](https://github.com/yiisoft/yii2-app-advanced/blob/master/docs/guide/README.md), each tier has its own root alias. For example,
-the front-end tier has a root alias `@frontend`, while the back-end tier root alias is `@backend`. As a result,
-you may put the front-end classes under the namespace `frontend` while the back-end classes are under `backend`. This will
-allow these classes to be autoloaded by the Yii autoloader.
+Note that test-specific requirements are in `autoload-dev` in order not to pollute autoloader in production where Composer
+is used with `--no-dev` flag.
 
-To add a custom namespace to the autoloader you need to define an alias for the base directory of the namespace using [[Yii::setAlias()]].
-For example to load classes in the `foo` namespace that are located in the `path/to/foo` directory you will call `Yii::setAlias('@foo', 'path/to/foo')`.
+See [Composer documentation](https://getcomposer.org/doc/01-basic-usage.md#autoloading) for details.
 
-Class Map <span id="class-map"></span>
----------
+After configuration is done, run `composer dump-autoload` to generate autoloader. Note that it should be done after each
+re-configuration.
 
-The Yii class autoloader supports the *class map* feature, which maps class names to the corresponding class file paths.
-When the autoloader is loading a class, it will first check if the class is found in the map. If so, the corresponding
-file path will be included directly without further checks. This makes class autoloading super fast. In fact,
-all core Yii classes are autoloaded this way.
-
-You may add a class to the class map, stored in `Yii::$classMap`, using:
+In order for your application to be aware of autoloader, require `vendor/autoload.php` in your entry script. For web
+application it is typically `index.php`, for console it's `yii`:
 
 ```php
-Yii::$classMap['foo\bar\MyClass'] = 'path/to/MyClass.php';
-```
+<?php
 
-[Aliases](concept-aliases.md) can be used to specify class file paths. You should set the class map in the
-[bootstrapping](runtime-bootstrapping.md) process so that the map is ready before your classes are used.
+// comment out the following two lines when deployed to production
+defined('YII_DEBUG') or define('YII_DEBUG', true);
+defined('YII_ENV') or define('YII_ENV', 'dev');
 
+require(__DIR__ . '/../vendor/autoload.php');
+require(__DIR__ . '/../vendor/yiisoft/yii2/Yii.php');
```

Comment:
```php
require(__DIR__ . '/../vendor/autoload.php');
```
should do the trick, no? Otherwise, a default `composer.json` should be able to load `Yii` as well.